yargs.js

```diff
@@ -330,7 +330,7 @@ function Yargs (processArgs, cwd, parentRequire) {
     argsert('[object|string] [string|function] [function]', [key, msg, parseFn], arguments.length)
     // allow a config object to be provided directly.
     if (typeof key === 'object') {
-      key = applyExtends(key, cwd)
+      key = applyExtends(key, cwd, self.getParserConfiguration()['merge-extends'])
```

Comment:
Is this the correct way to pass on configuration and if so, what name do you prefer :)

Reply:
As this is a yargs options, not related to the parser, I think you should not use `getParserConfiguration()`.

I suggest instead adding a `mergeExtends()` function to set a `mergeExtends` boolean variable (as `strict()` does for example).

Reply:
Yes, I did fully understood how everything work together, thanks for pointing me in the right direction!